### PR TITLE
🩹 buy remaining amounts for bids that exceed the current balance

### PR DESCRIPTION
**Pull request by Thomas-Smets**

### CODE REVIEW

**Review by acollette — APPROVED** [review]
lgtm

File: src/accounts/AccountV1.sol [hunk 1]
```diff
@@ -551,14 +551,36 @@ contract AccountV1 is AccountStorageV1, IAccount {
...
+        uint256[] memory assetTypes = IRegistry(registry).batchGetAssetTypes(assetAddresses);
+        uint256 balance;
+        for (uint256 i; i < assetAddresses.length; ++i) {
+            // Skip if amount is 0.
+            if (assetAmounts[i] == 0) continue;
+
+            if (assetTypes[i] == 1) {
+                balance = erc20Balances[assetAddresses[i]];
+            } else if (assetTypes[i] == 2) {
+                if (address(this) == IERC721(assetAddresses[i]).ownerOf(assetIds[i])) balance = 1;
```
Comment (j-vp) [inline src/accounts/AccountV1.sol:571]: if this is false, the balance from the previous loop is used and balance is not set to zero, can cause problems. Reset balance to zero on every loop

Comment (j-vp) [reply src/accounts/AccountV1.sol:571]: or add an else balance=0 here

**Review by j-vp — CHANGES_REQUESTED** [review]
see balance comment, blocking merge for clarity

File: src/accounts/AccountV1.sol (hunk 1 above)
Comment (Thomas-Smets) [reply src/accounts/AccountV1.sol:571]: Could indeed lead to unnecessary reverts, good catch.
I did not fuzz the order of assets, so balance was always 0 before the ERC721s in the tests.
image

**Pull request merged**